### PR TITLE
restructure component models

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
separate required vs model-specific (internal) functions and objects in component model files

follows off of #644 

## Content
- [x] distinguish between coupler-required and model-specific functions in component model files
- [x] merge bucket_init.jl and bucket_utils.jl
- [x] rename component model files to `<model>.jl`
- [x] alphabetize `get_field` and `update_field!` methods for easier comparison/readability

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 1]
```diff
@@ -110,17 +75,78 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
...
+function get_model_prog_state(sim::ClimaAtmosSimulation)
+    return sim.integrator.u
+end
+
+"""
+    get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:radiative_energy_flux_toa})
+
+Extension of Interfacer.get_field to get the net TOA radiation, which is a sum of the
+upward and downward longwave and shortwave radiation.
+"""
```
Comment (LenkaNovak) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:92]: I noticed recently that sometimes the docstrings of extensions can sometimes cause the extensions to be ignored. It doesn't seem to be problematic here, but we should probably decide on a convention (Clima wide) as to how to deal with this. (not as part of this PR)

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 2]
```diff
@@ -110,17 +75,78 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
...
+function get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:radiative_energy_flux_toa})
+    FT = eltype(atmos_sim.integrator.u)
+    # save radiation source
```
Comment (LenkaNovak) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:95]: redundant comment

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 3]
```diff
@@ -110,17 +75,78 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
...
+get_field(sim::ClimaAtmosSimulation, ::Val{:air_density}) =
+    TD.air_density.(thermo_params, sim.integrator.p.precomputed.ᶜts)
+get_field(sim::ClimaAtmosSimulation, ::Val{:air_temperature}) =
+    TD.air_temperature.(thermo_params, sim.integrator.p.precomputed.ᶜts)
+get_field(sim::ClimaAtmosSimulation, ::Val{:liquid_precipitation}) = sim.integrator.p.precipitation.col_integrated_rain # kg/m^2/s
 get_field(sim::ClimaAtmosSimulation, ::Val{:radiative_energy_flux_sfc}) =
     ClimaCore.Fields.level(sim.integrator.p.radiation.ᶠradiation_flux, half)
-get_field(sim::ClimaAtmosSimulation, ::Val{:liquid_precipitation}) = sim.integrator.p.precipitation.col_integrated_rain # kg/m^2/s
 get_field(sim::ClimaAtmosSimulation, ::Val{:snow_precipitation}) = sim.integrator.p.precipitation.col_integrated_snow  # kg/m^2/s
 get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_energy_flux}) =
```
Comment (LenkaNovak) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:143]: again we can remove the units here

File: experiments/AMIP/components/atmosphere/climaatmos.jl [hunk 4]
```diff
@@ -298,64 +359,6 @@ function calculate_surface_air_density(atmos_sim::ClimaAtmosSimulation, T_S::Cli
     extrapolate_ρ_to_sfc.(Ref(thermo_params), ts_int, swap_space!(ones(axes(ts_int.ρ)), T_S))
```
Comment (LenkaNovak) [inline experiments/AMIP/components/atmosphere/climaatmos.jl:359]: `calculate_surface_air_density` one is required

**Review by LenkaNovak — APPROVED** [review]
It looks like the rebase removed some of the changes in the previous PR, e.g. `Spaces -> ClimaCore.Spaces` , and perhaps some of the imports. It may be good to look over the changes [here](https://github.com/CliMA/ClimaCoupler.jl/pull/668/files#diff-459c85f7df7d2d54eb1fb4a9fd1d5c1f7aaae748bc77a474c201d9581f04b0e1) just to double check. It would be good to check that there are no behavior changes both for slabplanet and AMIP before merging. Otherwise LGTM, thank you for doing this!  Approving, but I'm happy to take a look again if needed.

File: experiments/AMIP/components/atmosphere/climaatmos.jl (hunk 1 above)
Comment (juliasloan25) [reply experiments/AMIP/components/atmosphere/climaatmos.jl:92]: good point!

File: experiments/AMIP/components/atmosphere/climaatmos.jl (hunk 4 above)
Comment (juliasloan25) [reply experiments/AMIP/components/atmosphere/climaatmos.jl:359]: Ah we missed this in the docs - I'll update them

Do we need to require both `calculate_surface_air_density` and `get_field(:air_density)`?

Comment (juliasloan25) [reply experiments/AMIP/components/atmosphere/climaatmos.jl]: update: as discussed offline, these two functions return different values, which the docs have been updated to reflect

**Pull request merged**